source/appModules/eclipse.py

```diff
@@ -17,7 +26,102 @@ def event_valueChange(self):
 		# Therefore, just drop this event.
 		pass
 
+	def event_caret(self):
+		super(EclipseTextArea, self).event_caret()
+
+		# Check suggestion item and close the list if it is not valid
+		try:
+			if not self.appModule.selectedItem.name:
+				self.event_suggestionsClosed()
+				self.appModule.selectedItem = None
+		except:
+			pass
+
+	@script(
+		gestures = ["kb:enter", "kb:escape"]
+	)
+	def script_closeAutocompleter(self, gesture):
+		gesture.send()
+		
+		# Close the suggestions list if it is opened
+		if self.appModule.selectedItem:
+			self.event_suggestionsClosed()
+			self.appModule.selectedItem = None
+
+	@script(
+		# Translators: Input help mode message for the 'read documentation script
+		description = _("Tries to read documentation for the selected autocompletion item."),
+		gesture = "kb:nvda+d"
+	)
+	def script_readDocumentation(self, gesture):
+		# If there aren't any suggestion selected, there is no way to find quick documentation
+		if not self.appModule.selectedItem:
+			gesture.send()
+			return
+
+		obj = self.appModule.selectedItem.parent.parent.parent.parent.previous.previous
+		while obj:
+			obj = obj.firstChild
+			
+			if obj and obj.role == controlTypes.ROLE_DOCUMENT:
+				break
+
+		if obj:
+			api.setNavigatorObject(obj)
+			braille.handler.handleReviewMove()
+			sayAllHandler.readText(sayAllHandler.CURSOR_REVIEW)
+
+		else:
+			# Translators: When the help popup cannot be found for the selected autocompletion item
+			ui.message(_("Cann't find the documentation window."))
+
+	@script(
+		gesture="kb:tab"
+	)
+	def script_completeInstruction(self, gesture):
+		"""
+		Performs a standard autocompletion with the `TAB` key.
+		"""
+
+		# We need to ensure that the autocompletion popup is open
+		if self.appModule.selectedItem:
+			self.script_closeAutocompleter(keyboardHandler.KeyboardInputGesture.fromName("enter"))
+			return
+
+		# If not, we send the 'tab' key as is
+		gesture.send()
+
+class AutocompletionListItem(IAccessible):
+
+	def event_selection(self):
+		if not self.appModule.selectedItem:
+			api.getFocusObject().event_suggestionsOpened()
+
+		# This is to ease finding the elp document
+		if self.appModule.selectedItem != self:
+			self.appModule.selectedItem = self
+
+		# When writing in the editor, the selection event is not fire.
+		# This is probably because the items are reused.
+		if self.appModule.selectedItemName != self.name:
+			self.appModule.selectedItemName = self.name
+			speech.cancelSpeech()
+
+			# Reporting as focused should be sufficient
+			self.reportFocus()
+
+			# Fixme: I picked up this from UIA SuggestionItem

```

Comment:
Please either remove this, or explain the problems that are caused by this code, and why it should be used for now.

Reply:
Addressed this, to.

Thanks again.